src/ucp/wireup/address.c

```diff
@@ -991,11 +950,9 @@ ucs_status_t ucp_address_unpack(ucp_worker_t *worker, const void *buffer,
         ++dev_index;
     } while (!last_dev);
 
-    ucs_assert((unsigned)(address - address_list) == address_count);
-
-out:
-    unpacked_address->address_count = address_count;
+    unpacked_address->address_count = address - address_list;
     unpacked_address->address_list  = address_list;
+out:
```

Comment:
minor: looks like no need in this label. Now used on line 898 only (can just return there)